File: algorithms/35. Search Insert Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if target in nums:
            return nums.index(target)
        else:
            length = len(nums)
            i = 0
            if target>nums[length-1]:
                return length
            elif target<nums[0]:
                return 0
            else:
                while i<length-1:
                    if nums[i]<target and nums[i+1]>target:
                        return i+1
                    else:
                        i+=1
                        

File: leetcode/35. Search Insert Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0, right = nums.size() -1;
        int mid = (left + right)/2;
        while(left < right){
            
            if(nums[mid] < target) left = mid + 1;
            else right = mid;
            mid = left + (right - left) / 2;
        }  
        return (nums.size()-1 == mid && target > nums[mid]) ? mid + 1 : mid;
    }
};